### DISCUSSION TITLE
Switching between sources.

### DISCUSSION THREAD

**Discussion by cristiAndreiTarasi** [ThibaultBee/StreamPack#166] [General]
Hi,

I was wondering if it’s possible to switch between streamers without breaking the current connection. For example, I start streaming video and audio, and at some point during the stream, I want to switch to streaming my device’s screen. Later, I want to switch back to the initial video/audio stream, all without interrupting, pausing, or breaking the existing connection.

I looked at the demos and noticed that screen recording is built as a separate feature from video streaming. Is this because StreamPack’s architecture doesn’t support the functionality I described, or was it done to better highlight the feature? Alternatively, is the screen recorder demo just recording the screen and saving it to the device?

Thank you.

**Comment by ThibaultBee** [comment]
You mean switching between audio and video input?

**Reply by cristiAndreiTarasi** [reply]
I should've explained it more clear. I was asking if StreamPack supports switching between different types of streams (like video/audio to screen sharing) within a single live stream session. Is it currently possible to switch between different streaming sources without stopping and starting a new stream? Can StreamPack be configured that way?

**Reply by ThibaultBee** [reply]
Ok, I see. You want to change the video source and/or the audio source on the fly.
Well, it is not possible yet. It is something I have in mind but it is not an easy task and I have got other prorities.

Anyway, switching the video input to the screen sharing might not be even possible because the screen sharing need to be in a service to run in background.

**Reply by cristiAndreiTarasi** [reply]
Ok, I understand. Thanks for clarification.

**Comment by ThibaultBee** [comment]
Work in progress.

**Reply by ThibaultBee** [reply]
And done since 3.0.0
